lib/analytics/index.js

```diff
@@ -0,0 +1,333 @@
+
+/** Parts of the code are taken from angular-cli code base which is governed by the following license:
+ * https://angular.io/license
+ */
+
+const os = require('os');
+const HttpRequest = require('../http/request.js');
+const uuid = require('uuid');
+const fs = require('fs');
+
+const path = require('path');
+const {execSync} = require('child_process');
+const {Logger, VERSION, fileExistsSync} = require('../utils');
+
+const GA_SERVER_URL = 'https://www.google-analytics.com';
+const GA_SERVER_PORT = '443';
+const GA_API_KEY = 'XuPojOTwQ6yTO758EV4hBg';
+const GA_TRACKING_ID = 'G-DEKPKZSLXS';
+
+const defaultSettings = {
+  enabled: false,
+  log_path: './logs/analytics',
+  client_id: uuid.v4()
+};
+
+const RESERVED_EVENT_NAMES = ['ad_activeview', 'ad_click', 'ad_exposure', 'ad_impression', 'ad_query', 'adunit_exposure',
```

Comment:
```suggestion
const ALLOWED_EVENT_NAMES = ['ad_activeview', 'ad_click', 'ad_exposure', 'ad_impression', 'ad_query', 'adunit_exposure',
```

Will help others understand what is allowed and what isn't

Reply:
These events are NOT allowed by google, they are reserved for their internal use

Reply:
Ok, thanks for the clarification.